src/main/java/io/vlingo/symbio/store/dispatch/Dispatchable.java

```diff
@@ -0,0 +1,51 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+
+package io.vlingo.symbio.store.dispatch;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+public abstract class Dispatchable {
```

Comment:
Also in a previous comment I may have stated that the `Dispatchable` must hold both the raw `State` and `Entry` items. This is true, but I was unclear as to how that works. The `StateStore` stores all of In the following: Native `Object` state as `State`, native `Source` as `Entry`, and `Metadata`, and if the `Metadata` is provided in the `write(..., metadata, ...)` it is attached to the raw `State`. The contract is that if the `Projection` uses the native `Object` state in the projection (easier and fast than going from native to `State` to native) then the contract is between the domain model (Aggregate) to provide its native state as the `object` in the `Metadata`. We can establish the same contract for `ObjectStore` and `Journal`, but at present `Metadata` is not provided as an optional parameter in any of the `persist()` and `append()` operations in the `ObjectStore` and `Journal`. This seems to be a serious omission and should be fixed. This would also require adding two columns for `Metadata` storage in both database types, just as is done in the `StateStore` implementations. Note that the native `Object` is never stored in the Metadata columns because it is already serialized as the `State`. We only persist the `value` and the `type` and those will be restored into the `Metadata` inside the `State` when read from the store.

Thus, I conclude that the contract should be:

- `Entry` must hold `Metadata` that may have the `List<Source<?>> sources` of all persisted `Source<?>` instances.

- Must dispatch both `State` and `List<Entry<?>` in the `Dispatchable`